src/Microsoft.Azure.EventHubs/EventHubClient.cs

```diff
@@ -219,19 +220,26 @@ public Task SendAsync(EventData eventData, string partitionKey)
         {
             // eventDatas null check is inside ValidateEvents
             int count = EventDataSender.ValidateEvents(eventDatas, null, partitionKey);
+
             EventHubsEventSource.Log.EventSendStart(this.ClientId, count, partitionKey);
+            Activity activity = EventHubsDiagnosticSource.StartSendActivity(this.ClientId, this.ConnectionStringBuilder, partitionKey, eventDatas, count);
+
+            Task sendTask = null;
             try
             {
-                await this.InnerSender.SendAsync(eventDatas, partitionKey).ConfigureAwait(false);
+                sendTask = this.InnerSender.SendAsync(eventDatas, partitionKey);
```

Comment:
Unfortunately, I have no background on eventhub.

What does it send and receive? Does it make sense to send Activity.Id, i.e. should it be encoded into payload/properties?

Reply:
so the ```EventData``` class, which is the payload contract has a ```Properties``` dictionary which could potentially be used here. 

@serkantkaraca - would that work? The reason to do this is to support cross-application correlation where producer and consumer(s) of given events could be linked